control_plane/attachment_service/src/tenant_state.rs

```diff
@@ -438,7 +445,49 @@ impl TenantState {
         // more work on the same pageservers we're already using.
         let mut modified = false;
 
+        // Remove any intent state that should no longer be present
```

Comment:
You're right.  These used to be separate because there was a generic build of "used_pageserver" between them, and we needed it to not include anything that was just removed.  But then that was simplified to use the secondary node list directly as the hard_excludes when scheduling, so these can be merged.

Reply:
73fd21979